File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        res = dict()

        for i in range(len(nums)):

            if (target - nums[i]) not in res:
                res[nums[i]] = i
            else:
                return [res[target - nums[i]], i]
            

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        int compliment;

        for(int i = 0; i< nums.length; i++){
            compliment = target - nums[i];
            if(seen.containsKey(compliment)){
                return new int[]{seen.getOrDefault(compliment, 0), i};
            }
            seen.put(nums[i], i);
        }
        return new int[]{};
    }
}